### DISCUSSION TITLE
Why are all the models and permissions loading...?

### DISCUSSION THREAD

**Discussion by MickProPay** [spatie/laravel-permission#2338] [Q&A]
Hi Guys, fairly new to the whole spatie permission package and it just doesn't make sense to me.

I currently have 161 permission and 15 roles and when i load my admin role management page there are more than 500 models loaded..
Why does spatie just not load the role / permissions the current use has...?

Video link
[All the model links](https://www.awesomescreenshot.com/video/15176949?key=805c9203c329ad628dbd9aa551054484)

So the 500 is when you start creating items on the site but even the base is loading everything. I have a Super admin setup and its a Role that has one permission that allows full access but why is everything being loaded!!??

**Comment by erikn69** [comment]
>  currently have 161 permission and 15 roles ......  are more than 500 models loaded

Hi, the math doesn't add up to me, wich version are you using? It only loads the role / permissions the current user has

**Reply by MickProPay** [reply]
I would think so yes but its loading and i can add a screen shot here.. 

Is there any steps you can recommend to trouble shoot with??

**Reply by erikn69** [reply]
I can't see in the video, It seems that I read 104 permissions,
also did you notice the sqls? the library loads all models from cache, not from db, from db just loads the role / permissions the current user has

**Reply by MickProPay** [reply]
Hi, can see there is a cach check and a Db query but the math doesnt add up.. even if its cached how do they get the values they get..?

**Reply by erikn69** [reply]
https://github.com/spatie/laravel-permission/blob/490fa00b0c8286659a534d822f7e74c684c86d27/src/PermissionRegistrar.php#L354-L366

**Comment by MickProPay** [comment]
My spatie package version 
name     : spatie/laravel-permission
descrip. : Permission handling for Laravel 6.0 and up
keywords : acl, laravel, permission, permissions, rbac, roles, security, spatie
versions : * 5.9.1
type     : library
license  : MIT License (MIT) (OSI approved) https://spdx.org/licenses/MIT.html#licenseText
homepage : https://github.com/spatie/laravel-permission
source   : [git] https://github.com/spatie/laravel-permission.git a88ed98c8937442737e0c50163682e832d608f13
dist     : [zip] https://api.github.com/repos/spatie/laravel-permission/zipball/a88ed98c8937442737e0c50163682e832d608f13 a88ed98c8937442737e0c50163682e832d608f13
path     : /home/vagrant/projects/snp-portal-solidarity/vendor/spatie/laravel-permission
names    : spatie/laravel-permission

support
issues : https://github.com/spatie/laravel-permission/issues
source : https://github.com/spatie/laravel-permission/tree/5.9.1

autoload
files
psr-4
Spatie\Permission\ => src

requires
illuminate/auth ^7.0|^8.0|^9.0|^10.0
illuminate/container ^7.0|^8.0|^9.0|^10.0
illuminate/contracts ^7.0|^8.0|^9.0|^10.0
illuminate/database ^7.0|^8.0|^9.0|^10.0
php ^7.3|^8.0

requires (dev)
orchestra/testbench ^5.0|^6.0|^7.0|^8.0
phpunit/phpunit ^9.4
predis/predis ^1.1
